### PR TITLE
Wire/issues/908

### PR DESCRIPTION
**Pull request by peter-lawrey**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/wire/JSONWire.java [hunk 1]
```diff
@@ -373,6 +395,30 @@ public void copyOne(@NotNull WireOut wire, boolean inMap, boolean topLevel) thro
...
+                in.readPosition(position);
+                return false;
+            }
+        }
+        int ch = in.peekUnsignedByte();
+        if (Character.isLetterOrDigit(ch)) {
+            in.readPosition(position);
+            return false;
+        }
+        while (ch > 0 && ch <= ' ') {
```
Comment (tgd) [inline src/main/java/net/openhft/chronicle/wire/JSONWire.java:414]: Q: if this is going to return true by this point why does the iteration continue?

File: src/main/java/net/openhft/chronicle/wire/JSONWire.java [hunk 2]
```diff
@@ -854,10 +900,17 @@ public JSONWire typeLiteral(@Nullable CharSequence type) {
         @Override
         public JSONValueOut typePrefix(@NotNull CharSequence typeName) {
             if (useTypes) {
-                startBlock('{');
+                boolean nested = bytes.peekUnsignedByte(bytes.writePosition() - 1) == '{';
```
Comment (tgd) [inline src/main/java/net/openhft/chronicle/wire/JSONWire.java:903]: Could `bytes.writePosition()-1` yield a value that breaks `peekUnsignedByte`?

**Review by tgd — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/wire/JSONWire.java (hunk 1 above)
Comment (peter-lawrey) [reply src/main/java/net/openhft/chronicle/wire/JSONWire.java:414]: ch = on line 416 takes the next character

File: src/main/java/net/openhft/chronicle/wire/JSONWire.java (hunk 2 above)
Comment (peter-lawrey) [reply src/main/java/net/openhft/chronicle/wire/JSONWire.java:903]: peekUnsignedByte returns -1 if out of bounds.

**Pull request merged**